### ISSUE TITLE
Operator CRDs

### ISSUE THREAD

**Issue by chrislovecnm** [cockroachdb/cockroach-operator#29]
As an operator, I can configure CockroachDB custom resources on a
new cluster in Kubernetes, so that I can initialize a cluster that will meet
my deployment needs.

**Comment by chrislovecnm** [comment]
Questions

1. Are we using CRDs for this?
2. If the operator does not own the cluster already how are we adjusting the settings?
3. How is number 4 and number 2 different?

**Comment by chrislovecnm** [comment]
https://github.com/chrislovecnm/cockroach-operator/blob/master/pkg/resource/statefulset.go seems to be where all of the configurations for the stateful set are located.

https://github.com/helm/charts/blob/master/stable/cockroachdb/templates/statefulset.yaml is a good example of the stateful set

**Comment by chrislovecnm** [comment]
https://www.cockroachlabs.com/docs/stable/cluster-settings.html does DevOps person use these settings?

> As an operator, I can configure CockroachDB custom resources on a
new cluster in Kubernetes, so that I can initialize a cluster that will meet
my deployment needs.

Where should we start? What components are not set?  Is this just command line configurations?

**Comment by johnrk** [comment]
@chrislovecnm ,
> Questions
> 
> 1. Are we using CRDs for this?
Yes, suspect the existing Operator's [CRD](https://github.com/cockroachdb/cockroach-operator/blob/master/config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml) contains our desired set of custom resources. @chrisseto , is this consistent with CockroachCloud?
> 2. If the operator does not own the cluster already how are we adjusting the settings?
I'm not sure about this one. We need to be able to configure cluster settings to be able to apply a license (["enterprise.license"](https://www.cockroachlabs.com/docs/stable/cluster-settings.html)).
> 3. How is number 4 and number 2 different?
#2 and #4 are different. #2 should be defined in the CRD and easily applied with yaml file like how one can configure "topology" in [3-nodes-insecure.yaml](https://github.com/chrislovecnm/cockroach-operator/blob/master/config/examples/3-nodes-insecure.yaml) because it's defined in our CRD. #4 is based on CockroachDB's Cluster which are not all defined in the CRD.

**Comment by johnrk** [comment]
@chrislovecnm ,

> https://www.cockroachlabs.com/docs/stable/cluster-settings.html does DevOps person use these settings?
> 
> > As an operator, I can configure CockroachDB custom resources on a
> > new cluster in Kubernetes, so that I can initialize a cluster that will meet
> > my deployment needs.
> 
> Where should we start? What components are not set? Is this just command line configurations?

We should start on making sure all of these custom resources in the [CRD](https://github.com/cockroachdb/cockroach-operator/blob/master/config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml) work. [nodes](https://github.com/cockroachdb/cockroach-operator/blob/master/config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml#L264), [version](https://github.com/cockroachdb/cockroach-operator/blob/master/config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml#L339), and [clientTLSSecret](https://www.cockroachlabs.com/docs/stable/cluster-settings.html) are good places to start because we need to be able to start secure clusters with the right version and multiple pods.

**Comment by johnrk** [comment]
I've separated out the CockroachDB Cluster Settings configuration in a separate issue: https://github.com/cockroachdb/cockroach-operator/issues/51
